### PR TITLE
AB#135852: Clear Response button 

### PR DESCRIPTION
**Pull request by MoizIqbal97**
This pr disables the clear response button during mandatory question

Changes Made:
- Button disabled depending on hasOptionalComponent
- Tooltip added
- Webhook notifcation icon set behind a feature flag

### CODE REVIEW

File: app/client-app/src/components/shared/SurveyPage.jsx [hunk 1]
```diff
@@ -168,17 +173,24 @@ const SurveyPage = ({
...
-            <Button
-              size="lg"
-              colorScheme="red"
-              variant="outline"
-              borderWidth="2px"
-              onClick={clearResult}
-              mr={2}
+            <Tooltip
+              label="This page requires a response before proceeding, so it cannot be cleared. You can continue to change your response before submitting."
+              isDisabled={hasOptionalComponent}
```
Comment (beforan) [inline app/client-app/src/components/shared/SurveyPage.jsx:178]: does this `isDisabled` control whether the tooltip shows?

Just to be sure:

it should only show if there are no optional components

if there are optional components, the button should work with no tooltip

File: app/client-app/src/components/shared/SurveyPage.jsx [hunk 2]
```diff
@@ -168,17 +173,24 @@ const SurveyPage = ({
...
-            </Button>
-            {webhookCount != null && (
+              <Button
+                size="lg"
+                colorScheme="red"
+                variant="outline"
+                borderWidth="2px"
+                onClick={clearResult}
+                mr={2}
+                disabled={!hasOptionalComponent}
```
Comment (beforan) [inline app/client-app/src/components/shared/SurveyPage.jsx:188]: I think Chakra has an `isDisabled` prop for buttons? disabled is the pure HTML way; I think `isDisabled` does extra stuff like setting the HTML attribute and also adding `aria-` attributes.

The `isDisabled` Button prop in the docs: https://chakra-ui.com/docs/components/button/props

**Review by beforan — CHANGES_REQUESTED** [review]

File: app/client-app/src/components/shared/SurveyPage.jsx [hunk 3]
```diff
@@ -168,17 +173,24 @@ const SurveyPage = ({
...
+              >
+                Clear Response
+              </Button>
+            </Tooltip>
+            {(webhookCount != null) & webhookEnabled && (
```
Comment (beforan) [inline app/client-app/src/components/shared/SurveyPage.jsx:193]:
```suggestion
            {(webhookCount != null) && webhookEnabled && (
```

**Review by beforan — CHANGES_REQUESTED** [review]

File: app/client-app/src/components/shared/SurveyPage.jsx (hunk 1 above)
Comment (MoizIqbal97) [reply app/client-app/src/components/shared/SurveyPage.jsx:178]: Yup, isDisabled={hasOptionalComponent} would make the tooltip not show during an optional component (when button is not disabled).

**Review by beforan — APPROVED** [review]

**Review by Jivanam — APPROVED** [review]

**Pull request merged**